extensions/stats/plugin.cc

```diff
@@ -137,13 +123,18 @@ const Node& NodeInfoCache::getPeerById(StringView peer_metadata_id_key,
   // Do not let the cache grow beyond max_cache_size_.
   if (cache_.size() > max_cache_size_) {
     auto it = cache_.begin();
-    for (int i = 0; i < int(max_cache_size_ / 4); ++i, ++it) {
-    }
-    cache_.erase(cache_.begin(), it);
+    cache_.erase(cache_.begin(), std::next(it, max_cache_size_ / 4));
     logInfo(absl::StrCat("cleaned cache, new cache_size:", cache_.size()));
   }
 
-  initializeNode(peer_metadata_key, &(cache_[peer_id]));
+  auto metadata = getMetadataStruct(MetadataType::Request, peer_metadata_key);
```

Comment:
`const auto& metadata`? SD filter is also written in this way, I need to change that as well...